### DISCUSSION TITLE
Speeding up propagation of css changes

### DISCUSSION THREAD

**Discussion by PlasmaHH** [squidfunk/mkdocs-material#7539] [Q&A]
It seems that css files are among the hardest to get a browser to reload, usually I personally resort to developer mode cache purge and reload to really get things out.

However I cannot expect from users of my pages to do that every time I want them to see changes to those files.

Now the file(s) whose changes I want people to quickly see is in extra_css and thus I thought that maybe there could be a feature  (optional) that adds a random string or so after ? to the url so that the browser does not cache that one.

**Comment by kamilkrzyskow** [comment]
Hello @PlasmaHH,
the cache busting feature was planned for 1.6.0 MkDocs then got moved to the 1.7.0 milestone and later got removed. So MkDocs doesn't have any plans to add it at the moment:
- https://github.com/mkdocs/mkdocs/issues/1979
- https://github.com/mkdocs/mkdocs/pull/3018

This can be achieved by using the [minify plugin](https://github.com/byrnereese/mkdocs-minify-plugin), where I added this feature a few versions back. I overcomplicated the solution a bit, as I wanted to get the hash from before the minification, but that's not needed.
- https://github.com/byrnereese/mkdocs-minify-plugin/pull/20

---

At [Nype](https://npe.cm/), minified and cache busted versions of the CSS files are [generated in the CI](https://github.com/nypesap/nypesap.github.io/blob/01f0f9df7abebb9c973ad9968c439c50b541bf34/.github/workflows/ci.yml#L31) via the execution of the [`__main__.py` file from the mkdocs-nype theme](https://github.com/nypesap/mkdocs-nype/blob/main/mkdocs_nype/__main__.py), this can be adapted into an external script in your project.

---

At [Privacy Guides](https://www.privacyguides.org/en/) they also simply use their own manually added `v=version` attribute:
- https://github.com/privacyguides/privacyguides.org/blob/b1c63004f262b6f5f8fbb006a5ee5a2889898179/mkdocs.yml#L285
